qutip/tests/test_correlation.py

```diff
@@ -251,3 +253,64 @@ def test_hamiltonian_order_unimportant():
     backwards = qutip.correlation_2op_2t(H[::-1], start, times, times, [sp],
                                          sp.dag(), sp)
     np.testing.assert_allclose(forwards, backwards, atol=1e-6)
+
+
+# Paramters for the test
+solver_list = ['me', 'es']
+start_list = [ 'coherent_dm', 'fock', None]
+e_op_list = ['non-hermitian', 'hermitian']
+w_list = [1,2] # Fast and slow oscillations
+gamma_list = [1,10] # Small and high damping
+args = list(product(solver_list, start_list, e_op_list, w_list, gamma_list))
+
+# We want to mark mc as slow
+solver_list = ['mc']
+start_list = [ 'fock']
+e_op_list = ['non-hermitian', 'hermitian']
+w_list = [1,2] # Fast and slow oscillations
+gamma_list = [1,10] # Small and high damping
+args_slow = list(product(solver_list, start_list, e_op_list, w_list, gamma_list))
+args += [pytest.param(*arg, marks = pytest.mark.slow) for arg in args_slow]
+
+
+@pytest.mark.filterwarnings("ignore::FutureWarning")
+@pytest.mark.parametrize(["solver", "start", "e_op", "w", "gamma"], args)
```

Comment:
First: what's issuing a `FutureWarning` in this test?  We probably should be using whatever it is.

Second: you can actually just use several calls to `pytest.mark.parametrize` to achieve this.  For example:
```python
@pytest.mark.parametrize(['solver', 'state'], [
    pytest.param('me', _equivalence_fock, id="me-ket"),
    pytest.param('me', _equivalence_dm, id="me-dm"),
    pytest.param('me', None, id="me-steady"),
    pytest.param('es', _equivalence_fock, id="es-ket"),
    pytest.param('es', _equivalence_dm, id="es-dm"),
    pytest.param('es', None, id="es-steady"),
    pytest.param('mc', _equivalence_fock, id="mc-steady", marks=[pytest.mark.slow]),
])
@pytest.mark.parametrize("hermitian", [True, False], ids=["hermitian", "nonhermitian"])
@pytest.mark.parametrize("w", [1, 2])
@pytest.mark.parametrize("gamma", [1, 10])
def test_known_case(...):
```

Reply:
> First: what's issuing a FutureWarning in this test? We probably shouldn't be using whatever it is.

The `FutureWarning` should not be there. I copied from another test that uses legacy code that does issue this `FutureWarning`. I will remove it.

> Second: you can actually just use several calls to pytest.mark.parametrize to achieve this. For example:

I did not know about this! Thank you, I will implement it in this way.